### DISCUSSION TITLE
Error appearing after Upgrading firebase and Vuefire packages

### DISCUSSION THREAD

**Discussion by AngryBacteria** [vuejs/vuefire#1489] [Questions and Help]
I am currently updating an older project of mine. It uses the following version of Vuefire and Firebase. With these versions everything works great:
- vuefire: 3.0.1
- firebase: 9.17.1

If I update the packages to their newest version on npm (3.1.22 and 10.8.0) I get an error with the `useFirebaseAuth` function. It only returns `undefined`. There is also an error that appears when trying to use that function. Below I pasted the error and the code block that is problematic. It is inside a Vue Component with the setup tag, which makes the error even weirder for me. I would greatly appreciate any help!

The error
```
[Vue warn]: inject() can only be used inside setup() or functional components. 
```

The codeblock (inside a vuejs 3 component with the setup function)
```javascript
async function login() {
  const auth = useFirebaseAuth();
  if (auth) {
    try {
      await signInWithEmailAndPassword(auth, email.value, password.value);
      Notify.create({
        message: 'Successfully logged in',
        color: 'primary',
        position: 'top',
        icon: 'mdi-check',
      });
      const to =
        route.currentRoute.value.query.redirect &&
          typeof route.currentRoute.value.query.redirect === 'string'
          ? route.currentRoute.value.query.redirect
          : '/';
      await router.push(to);
    } catch (error) {
      Notify.create({
        message: error ? `${error}` : 'Error while attempting to Log-In',
        color: 'red',
        position: 'top',
        icon: 'mdi-close-octagon-outline',
      });
    }
  }
}
```

**Comment by posva** [comment]
The warnings says it all:

```diff
+const auth = useFirebaseAuth();
 async function login() {
-  const auth = useFirebaseAuth();
   if (auth) {
```

this will work as long as you are within the `<script setup>`
